android/app/src/main/java/app/organicmaps/routing/RoutingController.java

```diff
@@ -198,7 +198,7 @@ private void processRoutingEvent()
 
   private boolean isDrivingOptionsBuildError()
   {
-    return !ResultCodesHelper.isMoreMapsNeeded(mLastResultCode) && RoutingOptions.hasAnyOptions();
+    return !ResultCodesHelper.isMoreMapsNeeded(mLastResultCode) && RoutingOptions.hasAnyOptions() && !isRulerRouterType();
```

Comment:
`isDrivingOptionsBuildError` looks confusing for the given issue, is it the right place to insert the check? When is `isDrivingOptionsBuildError` called? Maybe it's better to rename it? Or add the `isRulerRouterType` check outside of this function call?

Should a similar line in `android/app/src/main/java/app/organicmaps/routing/RoutingPlanController.java` also be updated?

Reply:
I also noticed the function name is a bit odd, but figured it should be the right place based on the fact a router filter was previously present here: https://github.com/organicmaps/organicmaps/pull/3601. I'd also agree to change its name to something more obvious; any suggestions?

As for calling in `RoutingPlanController.java`, I figured I could not reproduce a case where the warning showed, so I didn't change anything else. Will change.